include/envoy/stats/sink.h

```diff
@@ -40,6 +40,11 @@ class MetricSnapshot {
    * @return a snapshot of all text readouts.
    */
   virtual const std::vector<std::reference_wrapper<const TextReadout>>& textReadouts() PURE;
+
+  /**
+   * @return a snapshot of all text readouts.
+   */
+  virtual int64_t snapshotTimeMs() PURE;
```

Comment:
Can this return a proper chrono type and then be cast as needed elsewhere? Then you can name it `snapshotTime()` ?

Reply:
I commented about that as well, but I was thinking that @ramaraochavali wanted to factor out the conversion to milliseconds that must be done for each stat, and that this was (marginally) more efficient. There's a comment in the variable declaration.

It's fair to question this as premature optimization though....alternatively the comment about why we commit to ms in the API may also be good to express here in addition to at member variable declaration.

Also, nit: this method should be probably be const. Sorry I missed this before.

Reply:
Yes. Would like avoid that conversion if possible. Added the comment here and made const

Reply:
What conversion are we talking about? From an interface perspective can't you return a chrono and if perf matters that much you can convert in the impl above the loop body?

/wait-any

Reply:
Yes. We can do that.

Reply:
Changed. PTAL